server/src/test/scala/com/ossum/riddl/lsp/server/RiddlLSPTextDocumentSpec.scala

```diff
@@ -0,0 +1,142 @@
+    textDocumentIdentifier.setUri(docURI)
+
+    val service: RiddlLSPTextDocumentService = RiddlLSPTextDocumentService()
+  }
+
+  trait EmptyInitializeSpec {
+    val emptyDocURI: String = Path.of("server/src/test/resources/empty.riddl").toUri.toString
+    val textDocumentItem: TextDocumentItem = new TextDocumentItem()
+    textDocumentItem.setText("")
+    textDocumentItem.setUri(emptyDocURI)
+    val textDocumentIdentifier: TextDocumentIdentifier = new TextDocumentIdentifier()
+    textDocumentIdentifier.setUri(emptyDocURI)
+
+    val service: RiddlLSPTextDocumentService = RiddlLSPTextDocumentService()
+  }
+
+  trait OpenEverythingFileSpec extends EverythingInitializeSpec {
+    val openNotification: DidOpenTextDocumentParams = new DidOpenTextDocumentParams()
+    openNotification.setTextDocument(
+      textDocumentItem
+    )
+
+    service.didOpen(openNotification)
+  }
+
+  trait OpenEmptyFileSpec extends EmptyInitializeSpec {
+    val openNotification: DidOpenTextDocumentParams = new DidOpenTextDocumentParams()
+    openNotification.setTextDocument(
+      textDocumentItem
+    )
+
+    service.didOpen(openNotification)
+  }
+
+  trait ChangeEverythingFileSpec extends OpenEverythingFileSpec {
+    val changeNotification: DidChangeTextDocumentParams = new DidChangeTextDocumentParams()
+
+    val versionedDocIdentifier = new VersionedTextDocumentIdentifier()
+    versionedDocIdentifier.setUri(docURI)
+    changeNotification.setTextDocument(versionedDocIdentifier)
+
+    val changes = new TextDocumentContentChangeEvent()
+    changes.setText("the")
+    val changeRange = new lsp4j.Range()
+
+    val changeStart = new Position()
+    changeStart.setLine(11)
+    changeStart.setCharacter(21)
+    changeRange.setStart(changeStart)
+
+    val changeEnd = new Position()
+    changeEnd.setLine(11)
+    changeEnd.setCharacter(24)
+    changeRange.setEnd(changeEnd)
+
+    changes.setRange(changeRange)
+    changeNotification.setContentChanges(List(changes).asJava)
+
+    service.didChange(changeNotification)
+  }
+
+  trait ChangeEmptyFileSpec extends OpenEmptyFileSpec {
+    val changeNotification: DidChangeTextDocumentParams = new DidChangeTextDocumentParams()
+
+    val versionedDocIdentifier = new VersionedTextDocumentIdentifier()
+    versionedDocIdentifier.setUri(docURI)
+    changeNotification.setTextDocument(versionedDocIdentifier)
+
+    val changes = new TextDocumentContentChangeEvent()
+    changes.setText("domain New {}")
+    val changeRange = new lsp4j.Range()
+
+    val changeStart = new Position()
+    changeStart.setLine(1)
+    changeStart.setCharacter(1)
+    changeRange.setStart(changeStart)
+
+    val changeEnd = new Position()
+    changeEnd.setLine(1)
+    changeEnd.setCharacter(1)
+    changeRange.setEnd(changeEnd)
+
+    changes.setRange(changeRange)
+    changeNotification.setContentChanges(List(changes).asJava)
+
+    service.didChange(changeNotification)
```

Comment:
This is great and runs lots of things. But you need to put assertion tests in here to actually validate that it did things correctly.